### PR TITLE
Add an example for rust.nullable_field

### LINKED ISSUE #29: Create an example that uses `(rust.nullable_field)`
This task is to create an example in the `examples` crate that uses the (gogoproto.nullable) extension. Then update the main README to reflect the new example.

### PR DESCRIPTION
**Pull request by nipunn1313**
Fixes #29

### CODE REVIEW

**Comment by nipunn1313** [comment]
@ParkMyCar - how does it look?

File: examples/src/nullable_field/main.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+use proto_nullable_field::basic::{
+    Inner,
+    NonNullableMessage,
+};
+
+fn main() {
+    let _msg1 = NonNullableMessage {
```
Comment (ParkMyCar) [inline examples/src/nullable_field/main.rs:7]: Can you leave a comment here explaining how you used the rust.nullable_field option? Just something to link it back to the proto

**Review by ParkMyCar — APPROVED** [review]
Overall it looks good to me! Just one small nit w.r.t. adding a comment on the Rust side

File: examples/src/nullable_field/main.rs (hunk 1 above)
Comment (nipunn1313) [reply examples/src/nullable_field/main.rs]: Updated! Added comments and actually some asserts to show the difference in behavior (to help w/ questions like in #125)

**Pull request merged**